### PR TITLE
AdHash bid adapter: update to support latest version

### PR DESCRIPTION
**Pull request by wyand-sp**
Updating the Prebid.JS adaptor to support our platform version 3.2 and above.

## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
- Better support for brand-safety including the URL and the new formula
- Support for multiple top-level bidder URLs
- Use of local storage for recently viewed ads
- Support for ad-density

### CODE REVIEW

File: modules/adhashBidAdapter.js [hunk 1]
```diff
@@ -98,17 +148,50 @@ export const spec = {
...
     const bidRequests = [];
     let referrer = '';
-    if (bidderRequest && bidderRequest.refererInfo) {
-      // TODO: is 'page' the right value here?
-      referrer = bidderRequest.refererInfo.page;
+    try {
+      referrer = window.top.location.href;
+    } catch (e) {
+      referrer = window.location.href;
     }
```
Comment (Fawke) [inline modules/adhashBidAdapter.js:159]: You can read the referer value from bidderRequest object, `bidderRequest.refererInfo`

File: modules/adhashBidAdapter.js [hunk 2]
```diff
@@ -98,17 +148,50 @@ export const spec = {
...
+      body.scrollHeight,
+      body.offsetHeight,
+      html.clientHeight,
+      html.scrollHeight,
+      html.offsetHeight
+    );
+    const pageWidth = Math.max(body.scrollWidth, body.offsetWidth, html.clientWidth, html.scrollWidth, html.offsetWidth);
+
+    for (let i = 0; i < validBidRequests.length; i++) {
+      const bidderURL = validBidRequests[i].params.bidderURL || 'https://bidder.adhash.com';
```
Comment (Fawke) [inline modules/adhashBidAdapter.js:172]: Can you please ensure that the url passed through `params.bidderURL` is using `https://` and not `http://`.

**Review by Fawke — COMMENTED** [review]
Hi @wyand-sp,

I just added a few small comments, see if it's possible to work on those. Apart from that, it looks good.

File: modules/adhashBidAdapter.js (hunk 2 above)
Comment (wyand-sp) [reply modules/adhashBidAdapter.js:172]: I've added a check in isBidRequestValid as well as unit test for that.

File: modules/adhashBidAdapter.js (hunk 1 above)
Comment (wyand-sp) [reply modules/adhashBidAdapter.js:159]: I am now using bidderRequest.refererInfo.

**Review by Fawke — APPROVED** [review]

**Pull request merged**